src/axom/primal/operators/evaluate_integral.hpp

```diff
@@ -232,15 +172,16 @@ double evaluate_vector_line_integral(const primal::NURBSCurve<T, NDIMS>& n,
  *
  * \return the value of the integral
  */
-template <typename Lambda, typename T, int NDIMS>
-double evaluate_vector_line_integral(const axom::Array<primal::NURBSCurve<T, NDIMS>>& narray,
+template <typename Lambda, typename CurveType>
+double evaluate_vector_line_integral(const axom::Array<CurveType>& carray,
```

Comment:
Possible generalization for the future -- perhaps the exposed API should use `axom::ArrayView` instead of `axom::Array` ?

Reply:
I think this is a good idea, as we discussed earlier, but I'm also running into a case where it's useful to template the return type of the lambda function so that you can perform integrals on vector-type functions. Since `primal::Vector<T, NDIMS>` already support addition and scalar multiplication, it's a pretty straightforward addition, and would be really helpful for computing things like the centroid of geometric objects, since it reduces the overhead of evaluating the function for a single input double. I think I'm going to put both ideas in a new feature branch.